newton/_src/solvers/kamino/geometry/contacts.py

```
@@ -0,0 +1,715 @@
+        """
+        self._assert_has_data()
+        if self._data.num_model_max_contacts > 0:
+            self._data.reset()
+
+    ###
+    # Internals
+    ###
+
+    def _assert_has_data(self):
+        if self._data is None:
+            raise RuntimeError("ContactsData has not been allocated. Call `finalize()` before accessing data.")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)